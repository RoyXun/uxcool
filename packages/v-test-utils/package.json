{
  "name": "@suning/v-test-utils",
  "version": "1.0.10",
  "main": "src/index.js",
  "module": "src/index.js",
  "license": "MIT",
  "scripts": {
    "start": "npm run server",
    "compile": "v-tools run compile",
    "server": "v-tools run server"
  },
  "peerDependencies": {
    "vue": ">= 2.5.*"
  },
  "devDependencies": {
    "@suning/v-tools": "^1.0.35",
    "@vue/test-utils": "^1.0.0-beta.16"
  },
  "dependencies": {
    "@suning/v-trigger": "^1.0.64",
    "@vue/server-test-utils": "^1.0.0-beta.30",
    "jquery": "^3.3.1",
    "vue-server-renderer": "^2.6.11"
  }
}
