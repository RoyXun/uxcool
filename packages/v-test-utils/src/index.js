import Vue from 'vue';

import $ from 'jquery';
import { mount, createWrapper } from '@vue/test-utils';
import Trigger from '@suning/v-trigger';

export * from '@vue/test-utils';

// export * as server from '@vue/server-test-utils';

export { $, $ as jQuery };
// ms
export async function waitTime(delay = 20) {
  await new Promise((resolve) => {
    setTimeout(() => {
      resolve();
    }, delay);
  });
}

export function removePopup() {
  document.body.innerHTML = '';
}

const vueInst = new Vue();
export function getVueCreateElement() {
  return vueInst.$createElement;
}

/**
 * 获取Trigger实例
 * @param {Wrapper} wrapper
 */
export async function getPortal(wrapper) {
  const trigger = wrapper.find(Trigger);
  const portal = createWrapper(trigger.vm.portal.$el);
  return portal;
}

/**
 * 触发事件
 * @param {Wrapper} wrapper
 * @param {string} eventName
 * @param {number} delay
 */
export async function triggerEvent(wrapper, eventName, delay = 20) {
  wrapper.trigger(eventName);
  await waitTime(delay);
}

export function mountPickerFactory(Component) {
  return async (opts = {}) => {
    const wrapper = mount(Component, {
      ...opts,
    });
    await waitTime();
    return wrapper;
  };
}

export function destroyWrapper(wrapper) {
  wrapper.destroy();
}
