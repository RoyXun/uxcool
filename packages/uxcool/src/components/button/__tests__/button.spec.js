import { mountPickerFactory, waitTime } from '@suning/v-test-utils';
import Button from '../index';

describe('button', () => {
  const mountButton = mountPickerFactory(Button);
  // const mountButtonGroup = mountPickerFactory(Button.Group);
  it.each(['primary', 'ghost', 'dashed', 'danger', ''])(
    'render different buton types correctly',
    async (type) => {
      const wrapper = await mountButton({
        propsData: {
          type,
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    }
  );

  it.each(['small', 'default', 'large'])('render different sizes correctly', async (size) => {
    const wrapper = await mountButton({
      propsData: {
        size,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('emit event correctly', async () => {
    const clickFn = jest.fn();
    const wrapper = await mountButton({
      listeners: {
        click: clickFn,
      },
    });
    wrapper.trigger('click');
    expect(clickFn).toHaveBeenCalledTimes(1);
  });

  it('render loading correctly', async () => {
    const wrapper = await mountButton({
      propsData: {
        loading: { delay: 500 },
      },
      sync: false,
    });
    expect(wrapper.contains('.ux-btn-loading')).not.toBeTruthy();
    await waitTime(501);
    expect(wrapper.contains('.ux-btn-loading')).toBeTruthy();
    wrapper.destroy();
  });

  it('render loading changed correctly', async () => {
    const wrapper = await mountButton({
      propsData: {
        loading: { delay: 100 },
      },
      sync: false,
    });
    expect(wrapper.contains('.ux-btn-loading')).not.toBeTruthy();
    wrapper.setProps({ loading: { delay: 200 } });
    await waitTime(301);
    expect(wrapper.contains('.ux-btn-loading')).toBeTruthy();
    wrapper.destroy();
  });

  it.each(['submit', 'button', 'reset'])('render html type correctly', async (htmlType) => {
    const wrapper = await mountButton({
      propsData: {
        htmlType,
      },
    });
    expect(wrapper.find('button').element.getAttribute('type')).toBe(htmlType);
  });

  it('render link as button correctly', async () => {
    const wrapper = await mountButton({
      propsData: {
        href: 'http://www.suning.com',
      },
    });
    expect(wrapper.contains('button')).toBeFalsy();
    expect(wrapper.contains('a')).toBeTruthy();
    expect(wrapper.find('a').attributes('href')).toBe('http://www.suning.com');
    wrapper.setProps({ href: '' });
    await waitTime();
    expect(wrapper.contains('a')).toBeFalsy();
    expect(wrapper.contains('button')).toBeTruthy();
  });

  it('render button group correctly', async () => {
    const wrapper = await mountButton({
      render() {
        return (
          <Button.Group>
            <Button>a</Button>
            <Button>b</Button>
          </Button.Group>
        );
      },
    });
    const buttonGroup = wrapper.find('.ux-btn-group');
    expect(buttonGroup.findAll('button').length).toBe(2);
  });

  it('emit event correctly', async () => {
    const clickFn = jest.fn();
    const wrapper = await mountButton({
      listeners: {
        click: clickFn,
      },
    });
    wrapper.trigger('click');
    expect(clickFn).toHaveBeenCalledTimes(1);
    wrapper.destroy();
  });

  it('render button text is TwoCNChar correctly', async () => {
    const wrapper = await mountButton({
      render() {
        return (
          <div>
            <Button>中国</Button>
            <Button>
            <span>测试</span>
            </Button>
          <Button></Button>
          </div>
        );
      },
    });
    expect(wrapper.find('.ux-btn').text()).toBe('中 国');
    wrapper.destroy();
  });
});
