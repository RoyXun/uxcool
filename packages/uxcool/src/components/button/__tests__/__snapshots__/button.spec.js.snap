// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button render different buton types correctly 1`] = `
<button
  class="ux-btn ux-btn-primary"
  type="button"
/>
`;

exports[`button render different buton types correctly 2`] = `
<button
  class="ux-btn ux-btn-ghost"
  type="button"
/>
`;

exports[`button render different buton types correctly 3`] = `
<button
  class="ux-btn ux-btn-dashed"
  type="button"
/>
`;

exports[`button render different buton types correctly 4`] = `
<button
  class="ux-btn ux-btn-danger"
  type="button"
/>
`;

exports[`button render different buton types correctly 5`] = `
<button
  class="ux-btn"
  type="button"
/>
`;

exports[`button render different sizes correctly 1`] = `
<button
  class="ux-btn ux-btn-sm"
  type="button"
/>
`;

exports[`button render different sizes correctly 2`] = `
<button
  class="ux-btn"
  type="button"
/>
`;

exports[`button render different sizes correctly 3`] = `
<button
  class="ux-btn ux-btn-lg"
  type="button"
/>
`;
